doc/classes/Object.xml

```diff
@@ -76,8 +76,9 @@
 		<method name="_get_property_list" qualifiers="virtual">
 			<return type="Dictionary[]" />
 			<description>
-				Override this method to customize how script properties should be handled by the engine.
+				Override this method to provide a custom list of additional properties to handle by the engine.
```

Comment:
Not sure about saying "additional". It's misleading because they may also already exist, and the example below even contradicts it.

Reply:
If it could modify existing properties, `_validate_property()` would be pointless. I'll have to re-check, but the example is likely wrong 🙃

Reply:
Make no mistake, existing properties **created from a script**, not built-in

Reply:
Well the description is valid. This is what happens when you export `hammer_type`:
![image](https://github.com/godotengine/godot/assets/2223172/95054a82-7968-4b17-bf5c-3d76eb9d3dfd)
`_get_property_list()` still adds a property, even if it's a duplicate. Same applies to built-in properties, so they behave the same in this regard. The `hammer_type` added in the example is a new property, in addition to the existing one, which is not exposed in the inspector. The original property can't be modified using `_get_property_list()`.

Reply:
> _get_property_list() still adds a property, even if it's a duplicate. 

This was indeed the only way to go about this before `_validate_property()` was added, way after the example was written

Reply:
So should I clarify in the example that the other `hammer_type` is a new property? Or maybe rename one of them?
Or it's fine as is?

Reply:
In my opinion you could move the example to `_validate_property` and repurpose something else for `_get_property_list`, maybe inspired by the way TileMap layers are displayed or Mesh materials can be overridden. Not that the original example is inaccurate. I do repeat, it was the recommended and only way to do this before `_validate_property`.

Reply:
Pushed the new example. Here's how it looks when used:
![image](https://github.com/godotengine/godot/assets/2223172/95bd908f-0ccf-4e28-ab71-b5f44c035b19)
The C# example needs rewriting.

Reply:
```csharp
[Tool]
public partial class MyNode : Node
{
    private int _numberCount;

    [Export]
    public int NumberCount
    {
        get => _numberCount;
        set
        {
            _numberCount = value;
            _numbers.Resize(_numberCount);
            NotifyPropertyListChanged();
        }
    }

    private List<int> _numbers = new();

    public override Godot.Collections.Array<Godot.Collections.Dictionary> _GetPropertyList()
    {
        var properties = new Godot.Collections.Array<Godot.Collections.Dictionary>();

        for (int i = 0; i < _numberCount; i++)
        {
            properties.Add(new Godot.Collections.Dictionary()
            {
                { "name", $"number_{i}" },
                { "type", (int)Variant.Type.Int },
                { "hint", (int)PropertyHint.Enum },
                { "hint_string", "Zero,One,Two,Three,Four,Five" },
            });
        }

        return properties;
    }

    public override Variant _Get(StringName property)
    {
        string propertyName = property.ToString();
        if (propertyName.StartsWith("number_"))
        {
            int index = int.Parse(propertyName.Substring("number_".Length));
            return _numbers[index];
        }
        return default;
    }

    public override bool _Set(StringName property, Variant value)
    {
        string propertyName = property.ToString();
        if (propertyName.StartsWith("number_"))
        {
            int index = int.Parse(propertyName.Substring("number_".Length));
            numbers[index] = value.As<int>();
            return true;
        }
        return false;
    }
}
```